### ISSUE TITLE
[question] [feature request] How to "unblock"/passthrough certain keys to other extensions?

### ISSUE THREAD

**Issue by awillats** [Wojnach/vimflowy#26]
Thanks for the great extension,
I love using vim motions/keybings for workflowy, but I also want to use them elsewhere for browser navigation.
As such, I also use [vimium](https://github.com/philc/vimium) / [vimium-c](https://github.com/gdh1995/vimium-c) to navigate the browser. 

For the most part, I'm okay with just blacklisting workflowy URLs (as suggested in the README), but the real sticking point is I can't use my vim keybinds (e.g. J, K) to switch tabs _out of_ a workflowy.com tab, which interrupts my flow and forces me to use my mouse or one of the clunkier non-vim shortcuts for tab navigation.

Is there a way to selectively "unblock" / allow to pass-through a subset of keys in normal mode? (this would allow me to use keys unused by vimflowy but usable by vimium for navigation).
If needed, I'd be willing to just patch together changes to keybindings.js or transparentKeybindings.js on my own local fork as a workaround - I'm just not sure what I would need to do. Would setting "K" as a keymap which does nothing in transparentKeybindings be sufficient to let it pass through for instance?

as a more flexible future solution which would likely require a lot more work, it would be great to be able to specify keys that get passed through in the options tab - ( this is something vimium has implemented for example: [mode_key_handler.js](https://github.com/philc/vimium/blob/2c93c251118b43f447dbd0704f1ab04db466e2fd/content_scripts/mode_key_handler.js) )

**Comment by Wojnach** [comment]
Yeah, easiest way to fix it is to add bindings for the keys you want to unblock, and then leave them empty.

Ideally it would be done differently with a whitelist of keys perhaps as an option in the extension itself. But its more work than I have time for atm

//Sydney

On 17 Oct 2022, at 04:21, Adam Willats ***@***.***> wrote:

﻿

Thanks for the great extension,
I love using vim motions/keybings for workflowy, but I also want to use them elsewhere for browser navigation.
As such, I also use vimium<https://github.com/philc/vimium> / vimium-c<https://github.com/gdh1995/vimium-c> to navigate the browser.

For the most part, I'm okay with just blacklisting workflowy URLs (as suggested in the README), but the real sticking point is I can't use my vim keybinds (e.g. J, K) to switch tabs out of a workflowy.com tab, which interrupts my flow and forces me to use my mouse or one of the clunkier non-vim shortcuts for tab navigation.

Is there a way to selectively "unblock" / allow to pass-through a subset of keys in normal mode? (this would allow me to use keys unused by vimflowy but usable by vimium for navigation).
If needed, I'd be willing to just patch together changes to keybindings.js or transparentKeybindings.js on my own local fork as a workaround - I'm just not sure what I would need to do. Would setting "K" as a keymap which does nothing in transparentKeybindings be sufficient to let it pass through for instance?

as a more flexible future solution which would likely require a lot more work, it would be great to be able to specify keys that get passed through in the options tab - ( this is something vimium has implemented for example: mode_key_handler.js<https://github.com/philc/vimium/blob/2c93c251118b43f447dbd0704f1ab04db466e2fd/content_scripts/mode_key_handler.js> )
